src/commonMain/kotlin/exchange.dydx.abacus/calculator/MarginCalculator.kt

```diff
@@ -301,11 +302,37 @@ internal object MarginCalculator {
         return parser.asString(tradeInput?.get("marketId"))?.let { marketId ->
             val position = parser.asNativeMap(parser.value(subaccount, "openPositions.$marketId"))
             val currentSize = parser.asDouble(parser.value(position, "size.current")) ?: 0.0
-            val postOrderSize = parser.asDouble(parser.value(position, "size.postOrder")) ?: 0.0
+            val postOrderSize = tradeInput?.let { getPositionPostOrderSizeFromTrade(parser, tradeInput, currentSize) } ?: 0.0
             return postOrderSize.abs() - currentSize.abs()
         }
     }
 
+    /**
+     * @description Helper to determine post-order position size from current trade input instead of position.size.postOrder
+     * We need this estimate before the trade delta is applied to the position, as position post-order size may not be updated yet.
+     */
+    private fun getPositionPostOrderSizeFromTrade(
+        parser: ParserProtocol,
+        trade: Map<String, Any>,
+        currentPositionSize: Double,
+    ): Double {
+        val marketId = parser.asString(trade["marketId"])
+        val side = parser.asString(trade["side"])
+        val tradeSize = if (marketId != null && side != null) {
+            parser.asNativeMap(trade["summary"])?.let { summary ->
+                if (parser.asBool(summary["filled"]) == true) {
+                    val multiplier = if (side == "BUY") Numeric.double.NEGATIVE else Numeric.double.POSITIVE
+                    (parser.asDouble(summary["size"]) ?: Numeric.double.ZERO) * multiplier * Numeric.double.NEGATIVE
```

Comment:
qq: Why the `multiplier * NEGATIVE`? Why not just switch `multiplier` and remove this `NEGATIVE`?

Reply:
copied from delta code but u right!! we don't need to do that here, will fix